ci/gpu/build.sh

```diff
@@ -75,6 +75,19 @@ cd $WORKSPACE
 if hasArg --skip-tests; then
     logger "Skipping tests..."
 else
+    logger "Check GPU usage..."
+    nvidia-smi
+
+    # TODO
+    # logger "GoogleTests..."
+    # cd $WORKSPACE/cpp/build
+
+    # for gt in ${WORKSPACE}/cpp/build/gtests/* ; do
```

Comment:
hey @trxcllnt. Ops guy here. is there any reason these Google Tests were commented out?

Reply:
@ajschmidt8 ah shit, I forgot I'd done that locally. Currently CI isn't running the C++ GTests at all, so I copied this over from the cuDF script. Then I realized the C++ tests aren't currently passing for other reasons, and turning them on would fail CI for this PR, so they're commented out.

Reply:
@harrism do you know what we can do to fix the C++ tests? Currently the `CubicSplitTest` is failing with `RMM_ERROR_NOT_INITIALIZED`. If we can initialize RMM in those tests, I can enable C++ testing in CI here.

Reply:
`CubicSplineTest` should be working so I'm digging into that presently.

Reply:
I resolved the issue with RMM_ERROR_NOT_INITIALIZED in https://github.com/rapidsai/cuspatial/pull/175. Is that the only test that was failing?

Reply:
@thomcom thanks! Yes I think so. Once that's merged, I'll merge it in and uncomment this here.

Reply:
I think the correct thing to do is to change all non-legacy tests to not use rmmAlloc or rmmFree (use `rmm::device_buffer`, `rmm::device_vector`, and if necessary `device_memory_resource` classes).